### PR TITLE
Missing report web methods

### LINKED ISSUE #12523: #report_web_vuln and other #report_web_* methods missing for HTTP data service
It appears that `#report_web_vuln` and other `#report_web_*` methods are only available when using the database and not when using the HTTP data service.

## Steps to reproduce
As an example, I will use #12486 PR (scanner/http/host_header_injection aux module), which makes use of `#report_web_vuln`:
```
msf5 auxiliary(scanner/http/host_header_injection) > db_status
[*] Connected to remote_data_service: (https://localhost:5443). Connection type: http. Connection name: local-https-data-service.
msf5 auxiliary(scanner/http/host_header_injection) > run

[+] 1.2.3.4:80/drupal/ (1.2.3.4)(200)(GET)(evidence into body) is vulnerable to HTTP Host header injection
[-] Auxiliary failed: NoMethodError undefined method `report_web_vuln' for #<Metasploit::Framework::DataService::RemoteHTTPDataService:0x00007ff08da2b900>
Did you mean?  report_vuln
               report_web_site
[-] Call stack:
[-]   /r7/src/metasploit-framework/lib/metasploit/framework/data_service/proxy/core.rb:136:in `method_missing'
[-]   /r7/src/metasploit-framework/lib/msf/core/auxiliary/report.rb:363:in `report_web_vuln'
[-]   /r7/src/metasploit-framework/modules/auxiliary/scanner/http/host_header_injection.rb:99:in `run_host'
[-]   /r7/src/metasploit-framework/lib/msf/core/auxiliary/scanner.rb:111:in `block (2 levels) in run'
[-]   /r7/src/metasploit-framework/lib/msf/core/thread_manager.rb:106:in `block in spawn'
[*] Auxiliary module execution completed
```

## Expected behavior
```
msf5 auxiliary(scanner/http/host_header_injection) > db_status
[*] Connected to msf. Connection type: postgresql.
msf5 auxiliary(scanner/http/host_header_injection) > run

[+] 1.2.3.4:80/drupal/ (1.2.3.4)(200)(GET)(evidence into body) is vulnerable to HTTP Host header injection
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
```

### PR DESCRIPTION
**Pull request by dwelch-r7**
Resolves #12523 and #12334  

Adds missing apis for reporting web pages, forms and vulns

Still needs swagger docs but no point delaying review of the rest
## Verification

- [x] Start `msfconsole`
- [x] `use auxiliary/scanner/http/brute_dirs`
- [x] `set options`
- [ ] **Verify** it doesn't break when it discovers a directory

### CODE REVIEW

File: lib/msf/util/db_manager.rb [hunk 1]
```diff
@@ -37,12 +37,20 @@ def self.process_opts_workspace(opts, framework, required = true)
       raise ArgumentError.new("opts must include a valid :workspace")
     end
 
-    if wspace.kind_of?(String)
+    case wspace
```
Comment (busterb) [inline lib/msf/util/db_manager.rb]: Nice catch here, this seems pretty useful.

**Review by busterb — APPROVED** [review]
Checked it out locally, this looks like he right way to go. Thanks!

File: lib/msf/util/db_manager.rb (hunk 1 above)
Comment (busterb) [reply lib/msf/util/db_manager.rb]: @jmartin-r7 this might solve some of the Pro issues you were running into as well

Comment (jmartin-r7) [reply lib/msf/util/db_manager.rb]: This is one of the paths in the possible solutions set.  Definitely an improvement on detection.

There are other possible ramifications as this does not yet handle when the workspace name is `change` from under the existing object by another db connection.

**Comment by busterb** [comment]
## Release Notes

This adds web reporting methods to the Metasploit 5 data service, reenabling reporting support for several web vuln. modules with Metasploit 5.

**Pull request merged**